src/Razor/src/Microsoft.VisualStudio.Editor.Razor/RazorSolutionStatus.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.ComponentModel;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    internal abstract class RazorSolutionStatus
+    {
+        public abstract bool IsAvailable { get; }
+
+        public abstract event PropertyChangedEventHandler PropertyChanged;
```

Comment:
❓ Why is this abstract instead of virtual?

Reply:
It's the common pattern we typically utilize which is these contract types are just there for contracts and we choose to replicate all the members on consuming types. Inheritance has typically resulted in more pain than not when it comes to readability / understandability and rarely have we found benefit in utilizing base class members vs. just re-declaring.